hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloaderLock.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.nearcache.impl.preloader;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.logging.ILogger;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+
+import static com.hazelcast.nio.IOUtil.closeResource;
+
+class NearCachePreloaderLock {
```

Comment:
I notice two differences: `HazelcastException` instead of `HotRestartException` and I do a `lockFile.deleteOnExit();` in the `finally` block. This might either break some tests (different exceptions) or some behavior (lock file being deleted). Beside that we surely can use the same code base for both features.